### DISCUSSION TITLE
Editor mode like webflow

### DISCUSSION THREAD

**Discussion by PubliAlex** [webstudio-is/webstudio-community#160] [Wishlist]
Hello,

I wonder if webstudio plan to add an editor mode like Webflow. This is a really essential feature in many agencies (us included) to give opportunities to final clients to edit some texts or images without the possibility to destroy the layout or having to master a full no-code web development tool.

It would be a wonderful addition and the last hurdle to clear before we too could switch to webstudio.

**Comment by kof** [comment]
For now its a no. In the next 6 month we will be focused on building the best designer experience. Content editing is something CMS should do. Webstudio is not a CMS as opposed to webflow. We value the flexibility of the designer experience and we want the best possible content editing experience, which is not working great if you try to solve both problems in one tool.

**Reply by PubliAlex** [reply]
Just to be sure we talk about the same thing. I don't talk about dynamic content like blog post, products etc... that is a part of the CMS and it's great like that.

For something static, like a paragraph or an image somewhere on the home page, I'm not sure it merits the creation of an entity with fields, CRUD management etc... in any CMS.

It's like a very light version of the designer tool with most of the panels hidden. Just letting access to text and image edition. Just to be able to give access to final non-tech user without letting the ability to break the design completely.

**Reply by kof** [reply]
We are still thinking about it. Its not very useful to just let user edit text contents of elements but not able to rearrange and add elements. You basically need a rich text component which contains all the elements inside of it and becomes editable.

What bothers me personally is that:
1. Rich text component like webflow's is very limited to what you can add
2. If you allow editing using all instances there are no design constraints

What I am thinking is that we could add a block editor - a combination of both, ability to add any compnents and great editing experience:

1. Builder/designer defines block editor position and potentially limits what components can be used inside of it.
2. Customer can add allowed components and writes content comfortably. Similar to Notion editing experience.

**Reply by PubliAlex** [reply]
Hello, thank you for your answer. I will give you my 2 cent of a old web developper that work since 2006 and worked on hundreds of projects.

> We are still thinking about it. Its not very useful to just let user edit text contents of elements but not able to rearrange and add elements. You basically need a rich text component which contains all the elements inside of it and becomes editable.

Not agree with that point. When you create a simple showcase website for a non-technical end user, you don't want him to be able to touch anything other than the content or images. Many people have very poor taste in design and it sound a perfect balance to me to let the design stuff to the agency / designers, but allowing end user to change some content stuff. Basically, what you want to give access to is what you can find in the setting tab of webstudio (to handle image source, to handle link, etc...) + text edition and hide all the rest.

I remember a case where we gave access to customer to edit some part of the website with a richeditor with full option toolbar. One week later, he wrote all the texts using comic sans ms, in red, with a flashy yellow background because it's "more visible".... => never again

Putting whole section to a rich text in a CMS is a really bad solution to me for 2 reasons : 
- Customer can break the layout easily. Let's imagine you have 2 column section with image on the left and text on the right. No richeditor handle columns layout correctly and I pratically tested them all and I try to solve that problem since 15 years.
- In 2024, customer want to see the impact of it's change directly on frontend for that small changes. I can understand to have a separate admin panel when you write a blog post, a product sheet etc... on the CMS side and have to click a preview button to see the result. When you just want to change an image or rewrite a title on the frontpage, you just want to do it "while browsing", not in a separate admin area.

> What bothers me personally is that:
> 
> Rich text component like webflow's is very limited to what you can add

Completely agree with that plus it's not the role of richeditor to handle layout stuff.

> If you allow editing using all instances there are no design constraints

It depend of options you give access to, if you block drag & drop, and only give access to setting tab only, it can restrict access to content only modification.

> What I am thinking is that we could add a block editor - a combination of both, ability to add any compnents and great editing experience:
> 
> Builder/designer defines block editor position and potentially limits what components can be used inside of it.
> Customer can add allowed components and writes content comfortably. Similar to Notion editing experience.

The webflow approach is to provide a checkbox "editable in the editor" to all components, but adding a "editor" block and put what you want to make editable in it (a little like collection component if I understand well, a kind of "wrapper") sounds to be a completely awesome solution, but would require a lot of work. My first idea to just hide most stuff of the designer was made to minify the development need and provide an OK solution without waiting years, but your solution would be just awesome.

In any cases, this is a very useful feature and it you want to take some customers from webflow, you need to propose an equivalent.

**Reply by kof** [reply]
> Not agree with that point. When you create a simple showcase website for a non-technical end user, you don't want him to be able to touch anything other than the content or images. 

That includes adding content and rearranging content. That means access to adding instances, moving instances around.

> I remember a case where we gave access to customer to edit some part of the website with a richeditor with full option toolbar. One week later, he wrote all the texts using comic sans ms, in red, with a flashy yellow background because it's 

Our rich text has no access to font settings, only basic things. Style panel would be disabled in that case, so I don't know what exactly you disagree with, seems like misunderstanding.


> to see the impact of it's change directly on frontend for that small changes. 

So you would be ok if they could only change text and images of existing instances. Won't be able to ad a new instance of anything?


Allowing only edit text and images and only in cases where designer clicked: "Allow edit content" is doable and probably not very hard.

**Reply by PubliAlex** [reply]
> That includes adding content and rearranging content. That means access to adding instances, moving instances around.

Can be ok is it restrict complete layout break possibilities

> Our rich text has no access to font settings, only basic things. Style panel would be disabled in that case, so I don't know what exactly you disagree with, seems like misunderstanding.

Yes it's a misunderstanding, I was supposing the use of external CMS richeditor to edit some content as you suggested in your first answer.

> So you would be ok if they could only change text and images of existing instances. Won't be able to ad a new instance of anything?

Completely, at least, for a first iteration

> Allowing only edit text and images and only in cases where designer clicked: "Allow edit content" is doable and probably not very hard.

That was the objective of my suggestion. I saw your roadmap that is quite full and full of very good things, so the idea was to think about a solution, not development heavy. That's why my first thought was to just simplify the designer we already have by removing most of it's panels. If the feature is useful and loved, it can be improved later, but at least, it's not a missing feature that would prevent lot of potential webflow users to switch.

**Reply by kof** [reply]
Definitely worth considering, I will write up an issue and get feedback on it. My main concern with this is if enough of people going to be happy with a text and image only editing permissions that can be handed over to the customer. No additional instances etc.

**Reply by PubliAlex** [reply]
> Definitely worth considering, I will write up an issue and get feedback on it. My main concern with this is if enough of people going to be happy with a text and image only editing permissions that can be handed over to the customer. No additional instances etc.

Thank you for the feedback. As said, the idea is to fill this gap without doing heavy development. Can be improved later depending on the popularity of this feature. By the way, I really like your idea of "block editor component" in a future iteration. Can be a nice addition to edit a full section containing several components instead of check "allow content edition" in each components individually

**Reply by ajhoyhoy** [reply]
I agree, with the sugestions here 100%, not having a webflow like editor "mode" is hindering my ability to migrate my clients to Webstudio over webflow. Not all clients are young and tech savy but we want to be able to empower the non tech savy type to make simple text, cms content, image edits easily on the page without looking at the "back-end" Webflows editor mode is quite simple to use. If this is a feature by the end of 2024 it will be a game changer!

**Reply by Zmove** [reply]
Same here. Can't switch while I can't give a simple text / media edit panel to my clients

**Comment by johnsicili** [comment]
This is live! https://wstd.us/content-mode-demo
